core/trino-main/src/main/java/io/trino/operator/scalar/RowToRowCast.java

```diff
@@ -165,6 +165,7 @@ private static Class<?> generateRowCast(Type fromType, Type toType, FunctionDepe
 
         // Embed the MD5 hash code of input and output types into the generated class name instead of the raw type names,
         // which could prevent the class name from hitting the length limitation and invalid characters.
+        @SuppressWarnings("deprecation")
         byte[] md5Suffix = Hashing.md5().hashBytes((fromType + "$" + toType).getBytes(UTF_8)).asBytes();
```

Comment:
This is internal and doesn't need to use MD5, so we could switch to `goodFastHash`

Reply:
Created https://github.com/trinodb/trino/pull/12650